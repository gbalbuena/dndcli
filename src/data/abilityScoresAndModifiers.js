import yamlService from './yamlService';

async function getAll() {
  return yamlService.getData('abilityScoresAndModifiers.yaml');
}

export default {
  getAll
}
