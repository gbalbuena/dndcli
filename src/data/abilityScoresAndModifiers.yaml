- score: [1]
  modifier: −5

- score: [2, 3]
  modifier: −4

- score: [4, 5]
  modifier: −3

- score: [6, 7]
  modifier: −2

- score: [8, 9]
  modifier: −1

- score: [10, 11]
  modifier: 0

- score: [12, 13]
  modifier: 1

- score: [14, 15]
  modifier: 2

- score: [16, 17]
  modifier: 3

- score: [18, 19]
  modifier: 4

- score: [20, 21]
  modifier: 5

- score: [22, 23]
  modifier: 6

- score: [24, 25]
  modifier: 7

- score: [26, 27]
  modifier: 8

- score: [28, 29]
  modifier: 9

- score: [30]
  modifier: 10
