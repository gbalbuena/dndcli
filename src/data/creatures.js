import yamlService from './yamlService';

async function getAll() {
  return yamlService.getData('creatures.yaml');
}

export default {
  getAll
}
