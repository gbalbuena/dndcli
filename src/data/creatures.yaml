- name: Chicken
  abilityScores:
    strength: 3
    dexterity: 10
    constitution: 8
    intelligence: 2
    wisdom: 12
    charisma: 4
  armorClass: 10
  HitPoints: 1 (1d4 - 1)
  challenge: 0xp
