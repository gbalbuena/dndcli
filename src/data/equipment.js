import yamlService from './yamlService';

async function getAll() {
  return yamlService.getData('equipment.yaml');
}

export default {
  getAll
}
