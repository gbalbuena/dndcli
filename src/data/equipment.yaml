- name: Javelin
  description: Proficiency with a javelin allows you to add your proficiency bonus to the attack roll for any attack you make with it.
  type: Simple Melee Weapon
  Weight: 2lbs
  cost: 5sp
  damage: 1d6 piercing
  properties:
  - Thrown (range 30ft/120ft)
  tags:
  - damage
  - combat

- name: Longsword
  description: Proficiency with a longsword allows you to add your proficiency bonus to the attack roll for any attack you make with it.
  type: Martial Melee Weapon
  weight: 3lbs
  cost: 15gp
  damage: 1d8 slashing
  properties:
  - Versatile (1d10)
  tags:
  - damage
  - combat
