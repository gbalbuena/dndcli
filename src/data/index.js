import fs from 'fs'
import yaml from 'js-yaml'

async function getCollection(filename) {
  return yaml.safeLoad(fs.readFileSync(`${__dirname}/${filename}`, 'utf8'));
}

export default {
  getCollection
}
