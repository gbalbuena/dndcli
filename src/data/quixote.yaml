name: Sir Quixote de La Mancha

abilityScores:
  strength: 18
  dexterity: 9
  constitution: 16
  intelligence: 11
  wisdom: 13
  charisma: 14
