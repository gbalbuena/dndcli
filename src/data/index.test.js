import data from './'

describe('data', () => {
  test('ability-scores-and-modifiers', async () => {
    expect(await data.getCollection('ability-scores-and-modifiers.yaml')).toMatchSnapshot();
  })
})
