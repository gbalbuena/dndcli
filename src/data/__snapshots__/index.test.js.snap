// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data ability-scores-and-modifiers 1`] = `
Array [
  Object {
    "modifier": "−5",
    "score": Array [
      1,
    ],
  },
  Object {
    "modifier": "−4",
    "score": Array [
      2,
      3,
    ],
  },
  Object {
    "modifier": "−3",
    "score": Array [
      4,
      5,
    ],
  },
  Object {
    "modifier": "−2",
    "score": Array [
      6,
      7,
    ],
  },
  Object {
    "modifier": "−1",
    "score": Array [
      8,
      9,
    ],
  },
  Object {
    "modifier": 0,
    "score": Array [
      10,
      11,
    ],
  },
  Object {
    "modifier": 1,
    "score": Array [
      12,
      13,
    ],
  },
  Object {
    "modifier": 2,
    "score": Array [
      14,
      15,
    ],
  },
  Object {
    "modifier": 3,
    "score": Array [
      16,
      17,
    ],
  },
  Object {
    "modifier": 4,
    "score": Array [
      18,
      19,
    ],
  },
  Object {
    "modifier": 5,
    "score": Array [
      20,
      21,
    ],
  },
  Object {
    "modifier": 6,
    "score": Array [
      22,
      23,
    ],
  },
  Object {
    "modifier": 7,
    "score": Array [
      24,
      25,
    ],
  },
  Object {
    "modifier": 8,
    "score": Array [
      26,
      27,
    ],
  },
  Object {
    "modifier": 9,
    "score": Array [
      28,
      29,
    ],
  },
  Object {
    "modifier": 10,
    "score": Array [
      30,
    ],
  },
]
`;
